### PR TITLE
[MRG] REFACTOR: add test, doc, enhancement for databoard and ramp-database

### PR DESCRIPTION
**Pull request by glemaitre**
TODO:

- [x] The different tests need to remove the file `databoard.db` in all case
- [ ] Modify files such that they use `ramp_config` instead of `ramp_kits_dir` and `ramp_data_dir`

### CODE REVIEW

File: databoard/tests/test_utils.py [hunk 1]
```diff
@@ -30,3 +30,11 @@ def test_generate_passwords():
...
     os.remove(users_to_add_f_name)
     os.remove(password_f_name)
+
+
+def test_import_module_from_source():
+        module_path = os.path.dirname(__file__)
+        mod = utils.import_module_from_source(
+                os.path.join(module_path, 'local_module.py'), 'mod'
+        )
+        assert hasattr(mod, 'func_local_module')
```
Comment (aboucaud) [inline databoard/tests/test_utils.py]: Add comment to understand why this tests the import

**Comment by glemaitre** [comment]
@aboucaud @jorisvandenbossche it would be ok now.
Let's figure out the password failure

**Comment by aboucaud** [comment]
Alright, tests pass now. Merging.

File: databoard/testing.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    approve_user('test_user')
+    create_user(
+        name='test_iris_admin', password='test',
+        lastname='Admin', firstname='Iris',
+        email='iris.admin@gmail.com', access_level='user')
+
+
+def _setup_ramp_kits_ramp_data(problem_name):
+    # TODO: This function does not have a unit test but only used in
+    # integration testing.
```
Comment (jorisvandenbossche) [inline databoard/testing.py]: Isn't that OK enough?

File: databoard/testing.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+
+def add_problems():
+    problems = ['iris', 'boston_housing']
+    for problem_name in problems:
+        _setup_ramp_kits_ramp_data(problem_name)
+        add_problem(problem_name)
+
+
+# def _add_problem_and_event(problem_name, test_user_name):
```
Comment (jorisvandenbossche) [inline databoard/testing.py]: is this to be removed?

File: databoard/testing.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+def create_test_db():
+    """Create an empty test db and setup the files s.
+
+    The different settings should be set as environment variables. The
+    environment variables to set are:
+
+    * DATABOARD_STAGE: stage of the database. Need to be `'TESTING'` to use
+      this function;
+    * DATABOARD_USER: database login;
+    * DATABOARD_PASSWORD: database password;
```
[the remark this answers is not in the document]
Comment (glemaitre) [reply databoard/testing.py]: True it was the case when using the SQLlite

File: databoard/testing.py (hunk 3 above)
Comment (glemaitre) [reply databoard/testing.py]: I did not write yet the `add_event` yet.

**Pull request merged**